testing/fetcher.ts

```diff
@@ -230,6 +231,49 @@ export class AuthenticatingFetcher implements Fetcher {
           headers: {...headers, Authorization: `Basic ${encodedAuth}`},
         };
       }
+      case AuthenticationType.Custom: {
+        let urlWithSubstitutions = url;
+        let bodyWithSubstitutions = body;
+        let formWithSubstitutions = JSON.stringify(form);
+        let headersWithSubstitutions = JSON.stringify(headers);
+
+        const {params} = this._credentials as CustomCredentials;
+
+        Object.entries(params).forEach(([key, value]) => {
+          if (urlWithSubstitutions) {
+            urlWithSubstitutions = urlWithSubstitutions.replaceAll(
+              `{{${key}-${this._invocationToken}}}`,
```

Comment:
I would make a helper method for this, just to eliminate the risk of a typo in one but not all of these, that would be very hard to catch.

Reply:
done. Just for my info, is this just a testing file that is supposed to mimic what we do in the `coda` repo?

Reply:
Yeah this just mimics what's done on the `coda` side. It's used by the CLI simulator, so if you run `coda execute path.ts --fetch` it'll use this code when making fetcher requests from the CLI.

It's annoying that it's grown to have so much code duplication, but the alternative would be try to abstract this auth handling and centralize it all the SDK, and have the `coda` call it, but the `coda` side is different enough that it hasn't felt like it's been worth prioritizing that.

Reply:
ahh gotcha ok so it does get exposed to the user tho in some form. thanks for the context!